### PR TITLE
Create hosting-on-linux-server

### PR DESCRIPTION
**Pull request by ookamicodes**
This would be a good guide if someone needs a 24/7 hosting with good performance.

### CODE REVIEW

**Review by almeidx — CHANGES_REQUESTED** [review]
Missing the `.md` file extension

File: other-guides/hosting-on-linux-server [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Hosting on a Linux Server
+
+The best way to host your bot is on a Linux Server (or simply said: just a virtual server. Its also for bare metal servers).
```
Comment (kyranet) [inline other-guides/hosting-on-linux-server:3]: Linux Server != Virtual Server.

File: other-guides/hosting-on-linux-server [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+## Requirements
+
+- A Linux Server with a stable version, node, npm, git, pm2.
```
Comment (kyranet) [inline other-guides/hosting-on-linux-server:7]: Stable version?? What does that exactly mean?? You can recommend **Debian 10**, as it's a very stable, robust, and minimal Linux distro, best suited for servers.

File: other-guides/hosting-on-linux-server [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Hosting on a Linux Server
+
+The best way to host your bot is on a Linux Server (or simply said: just a virtual server. Its also for bare metal servers).
+
+## Requirements
+
+- A Linux Server with a stable version, node, npm, git, pm2. 
+- Github repository from your bot (privat).
```
Comment (kyranet) [inline other-guides/hosting-on-linux-server:8]: GitHub isn't needed, people can use other services (Docker, GitLab, BitBucket, SourceForge, Gitea, self-hosted...), or even, upload the files directly.

File: other-guides/hosting-on-linux-server [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+- SSH Client
+
+## Getting Started
+
+### Login
+
+The best way to do that is with root access. If its root access the username would be `root`.
```
Comment (kyranet) [inline other-guides/hosting-on-linux-server:15]: While the default user is `root`, it's very unwise to use it, as a vulnerability in your bot or one of its services can put the entire system at risk.

Please recommend creating a new user with a password different to `root`.

File: other-guides/hosting-on-linux-server [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+The first step you do is to login into your Linux Server. You can use a SSH Client, with Windows CMD `ssh (username)@(ip)` or on macOS ssh 192.168.2.30 -l (username)`
```
Comment (kyranet) [inline other-guides/hosting-on-linux-server:16]: Not necessary for Windows, you can set up `%UserProfile%/.ssh/config` with something such as:

```yaml
Host my-server
  HostName 192.168.1.1
  Port 22
  User my-user
```

And then when you do `ssh my-server`, it automatically logs in.

File: other-guides/hosting-on-linux-server [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+After you login in, clone your bot repository to your server. We use git for that.
+To clone your repository just type `git clone https://github.com/(username)/(rep-name).git`. Username is your github account name and rep-name the name of your repository.
+Now you need to go the file folder. Type `ls`to see dir of your server. If you found a folder with the name of the repository this should be your bot files folder.
+Now type `cd (folder-name)`to get there. Now type again `ls`. If you see the files from the bot your right. 
+
+### Starting Bot
+
+Now deleted the node_modules folder (if you have) with `rm -r node_modules`
+and type next `npm i`. You need to reinstall that because the Linux got a other compiler then your windows computer (this is because there are diffrent OS). After you installed node_modules,
```
Comment (kyranet) [inline other-guides/hosting-on-linux-server:28]: You should never commit `node_modules`, in fact, you must add it to `.gitignore`.

File: other-guides/hosting-on-linux-server [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+### Pm2 Features
+
+#### Logs
+
+To see the console logs you need to type `pm2 logs`.
+
+#### Monitoring
+
+To monitor your application type `pm2 monit`. You can see awesome things like CPU/RAM usage, process lists, logs and more.
```
Comment (kyranet) [inline other-guides/hosting-on-linux-server:50]: There are more alternatives to `pm2`, I don't agree a "hosting on a Linux Server" guide should be biased towards very specific tools.

**Review by kyranet — CHANGES_REQUESTED** [review]

**Comment by eslachance** [comment]
All in all there's too much wrong with this PR to be allowed. 
- You say "Linux" but there's a hundred distros so it's not precise enough
- There's too much bias towards specific tools
- There's a *ton* of very bad suggestions here like being root on everything or comitting your node_modules folder
- it's missing *actual* information on how to do things even if all the above were fixed.

Sorry, I can't accept this at all.

**Pull request closed without merging**